src/sections/create-collection/collection-form/CollectionFormHelper.ts

```diff
@@ -0,0 +1,214 @@
+
+        fields[normalizedFieldName] = {
+          include: true,
+          optionalOrRequired:
+            isFieldRequiredByDataverse && !isAConditionallyRequiredField ? 'required' : 'optional',
+          parentBlockName: block.name as CollectionFormMetadataBlock
+        }
+
+        if (field.childMetadataFields) {
+          Object.entries(field.childMetadataFields).forEach(([_key, childField]) => {
+            const normalizedFieldName = this.replaceDotWithSlash(childField.name)
+            const isChildFieldRequiredByDataverse = childField.isRequired
+            const isAConditionallyRequiredChildField = CONDITIONALLY_REQUIRED_FIELDS.includes(
+              childField.name
+            )
+
+            childFields[normalizedFieldName] = {
+              include: true,
+              optionalOrRequired:
+                isChildFieldRequiredByDataverse && !isAConditionallyRequiredChildField
+                  ? 'required'
+                  : 'optional',
+              parentBlockName: block.name as CollectionFormMetadataBlock
+            }
+          })
+        }
+      })
+    })
+
+    return {
+      ...fields,
+      ...childFields
+    }
+  }
+
+  public static formatCollectiontInputLevels(
+    collectionInputLevels: CollectionInputLevel[] | undefined
+  ): FormattedCollectionInputLevelsWithoutParentBlockName {
+    const result: FormattedCollectionInputLevelsWithoutParentBlockName = {}
+
+    if (!collectionInputLevels) {
+      return result
+    }
+
+    collectionInputLevels.forEach((level) => {
+      const { datasetFieldName, include, required } = level
+      const replaceDotWithSlash = (str: string) => str.replace(/\./g, '/')
+      const normalizedFieldName = replaceDotWithSlash(datasetFieldName)
+
+      result[normalizedFieldName] = {
+        include,
+        optionalOrRequired: required ? 'required' : 'optional'
+      }
+    })
+    return result
+  }
+
+  public static mergeBaseAndDefaultInputLevels(
+    baseInputLevels: FormattedCollectionInputLevels,
+    formattedCollectionInputLevels: FormattedCollectionInputLevelsWithoutParentBlockName
+  ): FormattedCollectionInputLevels {
+    const result: FormattedCollectionInputLevels = { ...baseInputLevels }
+
+    for (const key in formattedCollectionInputLevels) {
+      if (baseInputLevels[key]) {
+        result[key] = {
+          ...baseInputLevels[key],
+          ...formattedCollectionInputLevels[key],
+          parentBlockName: baseInputLevels[key].parentBlockName
+        }
+      }
+    }
+
+    return result
+  }
+
+  public static separateMetadataBlocksInfoByNames(
+    allMetadataBlocksInfo: ReducedMetadataBlockInfo[]
+  ): {
+    citationBlock: ReducedMetadataBlockInfo
+    geospatialBlock: ReducedMetadataBlockInfo
+    socialScienceBlock: ReducedMetadataBlockInfo
+    astrophysicsBlock: ReducedMetadataBlockInfo
+    biomedicalBlock: ReducedMetadataBlockInfo
+    journalBlock: ReducedMetadataBlockInfo
+  } {
```

Comment:
If the list of metadataBlocks is retrieved from the API in the future, will this hard-coded list have to be converted to something dynamic?

Reply:
Exactly, all parts of the code where I explicitly define this 6 metadata blocks will be dynamic.